### PR TITLE
Add Alpine Linux variant and update.sh (fix #5)

### LINKED ISSUE #5: Use a Dockerfile.template to generate both php variant
There isn't many differences between the apache and fpm version, and the entrypoint is identical.

Using a Dockerfile.template with an `update.sh` script to generate / update the variants would make maintenance much easier.

For reference:
- https://github.com/nextcloud/docker/blob/master/update.sh
- https://github.com/nextcloud/docker/blob/master/Dockerfile-debian.template

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

**Comment by J0WI** [comment]
@thomascube would be great to get this to Docker Hub

**Comment by madmath03** [comment]
@J0WI, @thomascube : To fix the conflicts, could the changes of #27 be merged to this PR ?

**Comment by J0WI** [comment]
I've just rebased this again. Would be great to get some feedback before there are new conflicts.

**Comment by madmath03** [comment]
@J0WI The latest commit for #32 which upgraded to 1.3.8 generated new conflicts. Can you fix them please ?

@thomascube When do you think we could expect this to be merged ? The templates format and alpine variant would be great. Is there any help that could be provided to review this ?

**Comment by J0WI** [comment]
@thomascube will this PR ever be reviewed/merged?

**Comment by madmath03** [comment]
@thomascube Any update on when we can expect this to be merged ? This would really improve to have an alpine variant and docker templates.
Please let us know if there is anything we can do to help merging this.

**Comment by thomascube** [comment]
@J0WI after a first review, I have a few questions and remarks:

* What is the `generate-stackbrew-library.sh` for?
* `update.sh` doesn't run on my Mac :-(
  -> `./update.sh: line 4: apache: unbound variable`
  on `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)`
  plus incompatible options on `grep` and `sed`

I'll try to provide suggestions how to fix the Mac issues

**Comment by J0WI** [comment]
Associative arrays are supported since Bash 4.0 (February 2009). Apple refuses to provide an update for MacOS, so you may consider installing one from MacPorts or Homebrew. Otherwise I'd welcome suggestions for a simple POSIX alternative.
I've updated `grep` and `sed` to be POSIX compatible.

`generate-stackbrew-library.sh` generates the file for the [docker-library](https://github.com/docker-library/official-images/tree/master/library). This script will require [bashbrew](https://github.com/docker-library/official-images/tree/master/bashbrew).

File: update.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ declare -A base=(
 latest="$(
 	git ls-remote --tags https://github.com/roundcube/roundcubemail.git \
 		| cut -d/ -f3 \
-		| grep -P -- '^[\d\.]+$' \
+		| grep -E -- '^([0-9]{1,}\.)+[0-9]{1,}$' \
 		| sort -V \
 		| tail -1
 )"
```
Comment (thomascube) [inline update.sh]: The latest stable version can be fetched from here:
```
latest="$(curl -s http://roundcube.net/VERSION.txt)"
```

Comment (J0WI) [reply update.sh]: Thanks, changed in https://github.com/roundcube/roundcubemail-docker/pull/20/commits/4bace5ffed724f23600e11a5aefcaf0a82eb98fd

**Comment by J0WI** [comment]
@thomascube I've rebased this again.

**Comment by compuguy** [comment]
Any updates on this @J0WI?

**Comment by J0WI** [comment]
Nope, still waiting for a review from @thomascube or @alecpl .

**Comment by madmath03** [comment]
Hello there!

@J0WI I do not know if you've seen it, but there are some conflicts on this PR.

@thomascube & @alecpl : I know you guys are busy and spending time on Open Source project is hard, but would you consider creating a team to help maintain this repository ?

This image is already at more than 800K downloads on DockerHub and still growing. Making it "cleaner" and THE official roundcube image would really be nice and I'm sure you could easily find motivated roundcube fans to support you (I'm certainly one of them).
Just to make things clear, I am NOT blaming you guys and I'm truly grateful for all you're doing for this community. I just want you to know there are people that could help you.
So if you're inclined to letting people give you a hand, do not hesitate to ask 😉

**Comment by J0WI** [comment]
@madmath03 I noticed the conflicts, but since there is no statement from @thomascube or @alecpl it's worth noting to rebase it every few months again.

**Comment by madmath03** [comment]
@thomascube : any news regarding this PR ?

**Comment by J0WI** [comment]
Rebased.

**Comment by thomascube** [comment]
It doesn't fully work on my Mac OS which is a requirement for me. Need some time to investigate...

**Comment by J0WI** [comment]
A assume you mean the `update.sh`? AFAIK macOS ships with an ancient version of bash that does not support associative arrays. You can use Homebrew or MacPorts to install a more recent version.

**Comment by J0WI** [comment]
Is there a chance to add this without `update.sh`?

**Comment by madmath03** [comment]
It would be a little hard to maintain all the variants without the update script, but I guess we could do that and open another PR with just the `update.sh`.
@thomascube Is that okay for you ?

**Comment by thomascube** [comment]
Works for me. I'll get back to test the update script with an upgraded bash on macOS as soon as possible.

**Comment by J0WI** [comment]
Just rebased this again.

**Comment by madmath03** [comment]
@J0WI Could remove the update.sh script from this PR and open a new one with it ?
Once it is done, I will merge it and @thomascube can update the DockerHub configuration (paths to Dockerfiles changing).

I can also provide you a PR with the docker-compose for #52 adapted to your branch.

**Comment by J0WI** [comment]
Done

**Comment by madmath03** [comment]
Thanks @J0WI 

@thomascube Now that this PR has been merged, can you update the DockerHub settings (Dockerfiles are stored in different folders).

**Comment by thomascube** [comment]
Dockerhub is not yet configured for automatic builds... but that's on my list anyway.

**Pull request merged**